src/test/kotlin/com/atlassian/performance/tools/lib/LicenseOverridingDatabase.kt

```diff
@@ -25,35 +30,29 @@ internal class LicenseOverridingDatabase(
         ssh: SshConnection
     ) {
         database.start(jira, ssh)
-        val licenseTable =when(database.getDbType())
-        {
-            DbType.MySql -> "jiradb.productlicense"
-            DbType.Postgres -> "productlicense"
+        val dbType = getDbType()
+        val licenseTable = when (dbType) {
+            MySql -> "jiradb.productlicense"
+            Postgres -> "productlicense"
         }
-        var client : SshSqlClient
-        when(database.getDbType()){
-            DbType.MySql -> client = SshMysqlClient()
-            DbType.Postgres -> client = SshPostgresClient()
-            else -> throw Exception("Unknow DB : ${database.getDbType()}")
+        val client = when (dbType) {
+            MySql -> SshMysqlClient()
+            Postgres -> SshPostgresClient()
         }
         client.runSql(ssh, "DELETE FROM $licenseTable;")
         logger.info("Licenses nuked")
         licenses.forEachIndexed { index, license ->
-            val flattenedLicense = license.lines().joinToString(separator = "") { it.trim() }
-
-            when(database.getDbType())
-            {
-                DbType.MySql -> client.runSql(
-                    ssh = ssh,
-                    sql = "INSERT INTO $licenseTable VALUES ($index, \"$flattenedLicense\");"
-                )
-                DbType.Postgres -> client.runSql(
-                    ssh = ssh,
-                    sql = "INSERT INTO $licenseTable (id, license) VALUES ($index, '$flattenedLicense');"
-                )
+            val flatLicenseText = license.readLines().joinToString(separator = "") { it.trim() }
+            val insert = when (dbType) {
+                MySql -> "INSERT INTO $licenseTable VALUES ($index, \"$flatLicenseText\");"
+                Postgres -> "INSERT INTO $licenseTable (id, license) VALUES ($index, '$flatLicenseText');"
             }
-            logger.info("Added license: ${license.substring(0..8)}...")
+            val insertFile = Files.createTempFile("license-insert", ".sql").toFile()
```

Comment:
Potentially/probably overkill, but calling insertFile.delete() or insertFile.deleteOnExit() would just ensure there are no duplicate copies of the license file lying around.

Reply:
Good call!

Reply:
What a great call. Defense in depth 👍